File: Main2_91.java
import java.util.Scanner;

public class Main2_91 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int H = sc.nextInt();
        int[] A = new int[H + 1];

        for (int i = 1; i <= H; i++) {
            A[i] = sc.nextInt();
        }

        for (int i = 1; i <= H; i++) {
            StringBuilder sb = new StringBuilder();
            sb.append("node ").append(i).append(": key = ").append(A[i]);

            if (i != 1) {
                sb.append(", parent key = ").append(A[i / 2]);
            }

            int left = 2 * i;
            if (left <= H) {
                sb.append(", left key = ").append(A[left]);
            }

            int right = 2 * i + 1;
            if (right <= H) {
                sb.append(", right key = ").append(A[right]);
            }

            sb.append(", ");
            System.out.println(sb.toString());
        }

        sc.close();
    }
}

File: Main2_91.py
import sys

def main():
    input = sys.stdin.readline

    H = int(input())
    A = [0] + list(map(int, input().split())) 

    for i in range(1, H + 1):
        output = f"node {i}: key = {A[i]}"

        if i != 1:
            output += f", parent key = {A[i // 2]}"

        left = 2 * i
        if left <= H:
            output += f", left key = {A[left]}"

        right = 2 * i + 1
        if right <= H:
            output += f", right key = {A[right]}"

        output += ","
        print(output)


if __name__ == "__main__":
    main()